### PR TITLE
Added Checkmarx One Parser

### PR DESCRIPTION
**Pull request by FelixHernandez**
## Added Checkmarx one parser
[sc-2313]

**Description**
Added Checkmarx One parser, Docs and unittests. `DEDUPE_ALGO_UNIQUE_ID_FROM_TOOL` is being use for dedupe.

### CODE REVIEW

**Review by mtesauro — APPROVED** [review]
Approved

**Review by grendel513 — APPROVED** [review]

**Review by cneill — CHANGES_REQUESTED** [review]
Noticed that this isn't capturing CWE IDs on Findings when they're provided by the tool

File: unittests/scans/checkmarx_one/many_findings.json [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                        "fullName": "com.example.api.integrationtests.utils.IntegratHelper.getClient.testPassword",
+                        "typeName": "char",
+                        "methodLine": 35,
+                        "definitions": "1"
+                    }
+                ]
+            },
+            "comments": {},
+            "vulnerabilityDetails": {
+                "cweId": 244,
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply unittests/scans/checkmarx_one/many_findings.json:216]: Sure. Working on it.

[the remark this answers is not in the document]
Comment (FelixHernandez) [reply unittests/scans/checkmarx_one/many_findings.json:216]: https://github.com/DefectDojo/django-DefectDojo/pull/9715/commits/447de674cfc5c42898f86917ac4be51eddc759d3 Solved

**Review by Maffooch — APPROVED** [review]

File: dojo/tools/checkmarx_one/parser.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    def get_findings(self, file, test):
+        data = json.load(file)
+        findings = []
+        if "vulnerabilities" in data:
+            results = data.get("vulnerabilities", [])
+            for result in results:
+                id = result.get("identifiers")[0].get("value")
+                cve = None
+                if 'vulnerabilityDetails' in result:
+                    cve = result.get("vulnerabilites").get("cweId")
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply dojo/tools/checkmarx_one/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9715/commits/aa95f47a9977885d1c7e2f2aeb6fcde522ed9e29

File: dojo/tools/checkmarx_one/parser.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                severity = result.get("severity")
+                locations_uri = result.get("location").get("file")
+                locations_startLine = result.get("location").get("start_line")
+                locations_endLine = result.get("location").get("end_line")
+                finding = Finding(
+                    unique_id_from_tool=id,
+                    file_path=locations_uri,
+                    title=id + "_" + locations_uri,
+                    test=test,
+                    cve=cve,
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply dojo/tools/checkmarx_one/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9715/commits/aa95f47a9977885d1c7e2f2aeb6fcde522ed9e29

File: dojo/tools/checkmarx_one/parser.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+                )
+                findings.append(finding)
+        elif "results" in data:
+            results = data.get("results", [])
+            for vulnerability in results:
+                result_type = vulnerability.get("type")
+                date = self._parse_date(vulnerability.get("firstFoundAt"))
+                cve = None
+                if 'vulnerabilityDetails' in vulnerability:
+                    cve = vulnerability.get("vulnerabilites", {}).get("cweId")
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply dojo/tools/checkmarx_one/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9715/commits/aa95f47a9977885d1c7e2f2aeb6fcde522ed9e29

File: dojo/tools/checkmarx_one/parser.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+                if result_type == "sast":
+                    descriptionDetails = vulnerability.get("description")
+                    file_path = vulnerability.get("data").get("nodes")[0].get("fileName")
+                    finding = Finding(
+                        description=descriptionDetails,
+                        title=descriptionDetails,
+                        file_path=file_path,
+                        date=date,
+                        cve=cve,
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply dojo/tools/checkmarx_one/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9715/commits/aa95f47a9977885d1c7e2f2aeb6fcde522ed9e29

File: dojo/tools/checkmarx_one/parser.py [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+                        )
+                    findings.append(finding)
+                if result_type == "kics":
+                    description = vulnerability.get("description")
+                    file_path = vulnerability.get("data").get("filename")
+                    finding = Finding(
+                        title=f'{description}',
+                        description=description,
+                        date=date,
+                        cve=cve,
```
[the remark this answers is not in the document]
Comment (FelixHernandez) [reply dojo/tools/checkmarx_one/parser.py]: https://github.com/DefectDojo/django-DefectDojo/pull/9715/commits/aa95f47a9977885d1c7e2f2aeb6fcde522ed9e29

**Review by cneill — APPROVED** [review]

**Pull request merged**